plugins/secrets/get.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const helper = require('./helper');
+const getSchema = schema.models.secret.get;
+const idSchema = joi.reach(schema.models.secret.base, 'id');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/secrets/{id}',
+    config: {
+        description: 'Get a single secret',
+        notes: 'Returns a secret record',
+        tags: ['api', 'secrets'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const userFactory = request.server.app.userFactory;
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const secretFactory = request.server.app.secretFactory;
+            const username = request.auth.credentials.username;
+
+            return secretFactory.get(request.params.id)
```

Comment:
Right. We didn't do that because we were handling the "build" case as well. But since we removed it we should just use Promise.all here